x-pack/filebeat/input/azureeventhub/input.go

```diff
@@ -14,25 +14,64 @@ import (
 	"sync"
 	"time"
 
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	"github.com/Azure/azure-event-hubs-go/v3/eph"
+	"github.com/mitchellh/hashstructure"
+
 	"github.com/elastic/beats/v7/filebeat/channel"
 	"github.com/elastic/beats/v7/filebeat/input"
 	"github.com/elastic/beats/v7/libbeat/beat"
 	conf "github.com/elastic/elastic-agent-libs/config"
 	"github.com/elastic/elastic-agent-libs/logp"
 	"github.com/elastic/elastic-agent-libs/mapstr"
-
-	eventhub "github.com/Azure/azure-event-hubs-go/v3"
-	"github.com/Azure/azure-event-hubs-go/v3/eph"
 )
 
 const (
 	eventHubConnector        = ";EntityPath="
 	expandEventListFromField = "records"
+	inputName                = "azure-eventhub"
 )
 
+func init() {
+	err := input.Register(inputName, NewInput)
+	if err != nil {
+		panic(fmt.Errorf("failed to register %v input: %w", inputName, err))
+	}
+}
+
+// configID computes a unique ID for the input configuration.
+//
+// It is used to identify the input in the registry and to detect
+// changes in the configuration.
+//
+// We will remove this function as we upgrade the input to the
+// v2 API (there is an ID in the v2 context).
+func configID(config *conf.C) (string, error) {
+	var tmp struct {
+		ID string `config:"id"`
+	}
+	if err := config.Unpack(&tmp); err != nil {
+		return "", fmt.Errorf("error extracting ID: %w", err)
+	}
+	if tmp.ID != "" {
+		return tmp.ID, nil
+	}
+
+	var h map[string]interface{}
+	_ = config.Unpack(&h)
+	id, err := hashstructure.Hash(h, nil)
+	if err != nil {
+		return "", fmt.Errorf("can not compute ID from configuration: %w", err)
+	}
+
+	return fmt.Sprintf("%16X", id), nil
+}
+
 // azureInput struct for the azure-eventhub input
 type azureInput struct {
+	id           string           // id of the input
```

Comment:
I'd note that this is only for metrics identification so that it is removed when the change to v2 is made. Probably move it and metrics to the bottom of the struct as a pair so their association is visible.